plugins/rule_engines/irods_rule_language/src/configuration.cpp

```diff
@@ -418,23 +424,56 @@ int hash_rules(const std::vector<std::string> &irbs, const int pid, std::string
     return 0;
 }
 
-class make_copy {
-public:
-        make_copy(const std::vector<std::string>& _irbs, const int _pid) : irbs_(_irbs), pid_(_pid) {
-            for(auto const &irb : _irbs) {
-                boost::filesystem::copy_file(get_rule_base_path(irb), get_rule_base_path_copy(irb, _pid));
-            }
-        }
-        ~make_copy() {
-            for(auto const &irb : irbs_) {
-                boost::filesystem::remove(get_rule_base_path_copy(irb, pid_));
-            }
+class in_memory_rulebases
+{
+  public:
+    in_memory_rulebases(const std::vector<std::string>& _irods_rule_bases)
+    {
+        for (auto const& irb : _irods_rule_bases) {
+            std::ifstream ifs(get_rule_base_path(irb));
+            rule_bases_.insert(
+                {irb, std::string({std::istreambuf_iterator<char>(ifs), std::istreambuf_iterator<char>()})});
         }
-private:
-        const std::vector<std::string> irbs_;
-        const int pid_;
+    }
+
+    const std::string& get_rulebase(const std::string& irb)
+    {
+        return rule_bases_.at(irb);
+    }
+
+  private:
+    std::unordered_map<std::string, std::string> rule_bases_;
 };
 
+int hash_rules_with_copy(const std::vector<std::string>& irods_rule_bases,
+                         std::string& digest,
+                         in_memory_rulebases& copy)
+{
+    irods::Hasher hasher;
+    irods::error ret = irods::getHasher("md5", hasher);
+    if (!ret.ok()) {
+        int status = ret.code();
+        logger::rule_engine::error("hash_rules_with_copy: cannot get hasher, status = {}", status);
+        return status;
+    }
+
+    for (auto const& irb : irods_rule_bases) {
+        try {
+            hasher.update(copy.get_rulebase(irb));
+        }
+        catch (std::out_of_range& ex) {
+            logger::rule_engine::warn("hash_rules_with_copy: attempted to access nonexistent rulebase [{}]", irb);
+        }
+    }
+
+    hasher.digest(digest);
+
+    logger::rule_engine::debug(
+        "hash_rules_with_copy: rulebases = {}; digest = {}", fmt::join(irods_rule_bases, ","), digest);
```

Comment:
Does `irods_rule_bases` hold filenames or the rule code of each file?

If it holds rule code, then let's change this to list only the filenames. Syslog will truncate the message if it exceeds the max message size.

Reply:
In this context, it should be just rulebase names (e.g. `core`), so it should be fine.

Reply:
Good.